### ISSUE TITLE
Change format for editable date and date ranges

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy#1764]
### Description

The goal would be to answer/extend [this need](https://github.com/Avaiga/taipy/issues/1037) even for editable date. The goal is not for display purposes but to change the order from MM/DD/YYYY to DD/MM/YYYY or more.

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by FredLL-Avaiga** [comment]
works for me (tm)

**Comment by FlorianJacta** [comment]
What do you mean?

**Comment by FredLL-Avaiga** [comment]
I mean I tried it and it works ie you can edit a date with the specified format

**Comment by ShootingStarD** [comment]
> I mean I tried it and it works ie you can edit a date with the specified format

Could you provide the code and taipy version used?

**Comment by ShootingStarD** [comment]
Here is the code that I tried on Taipy 3.1.1 Taipy GUI 3.1.4
```python
from taipy.gui import builder as tgb
from taipy.gui import Gui
import datetime as dt
date_range = [dt.datetime(2024,1,5), dt.datetime(2024,1,10)]
with tgb.Page() as page:
    tgb.date_range(dates="{date_range}", format="dd-MM-yyyy")
    
Gui(page,).run(debug=True, use_reloader=True, port=5111)
```

![image](https://github.com/user-attachments/assets/cfaf0fbe-8dc9-4005-a560-b227e9c72e6c)

**Comment by FredLL-Avaiga** [comment]
Tried it on the develop branch (ie 4.0.0)

**Comment by ShootingStarD** [comment]
Indeed, sorry I didn't checked the dev version.

**Comment by jrobinAV** [comment]
Implemented in 4.0.
